neuralmonkey/learning_utils.py

```diff
@@ -137,7 +137,10 @@ def training_loop(tf_manager: TensorFlowManager,
         tf_manager.initialize_model_parts(
             runners + [trainer], save=True)  # type: ignore
     else:
-        tf_manager.restore(initial_variables)
+        try:
+            tf_manager.restore(initial_variables)
+        except tf.errors.NotFoundError:
+            warn("Some variables were not found in checkpoint.)")
```

Comment:
Will this leave the model randomly initialized or does it initialize the existing variables and leave uninitialized just the nonexistent ones?

Reply:
It's only relevant in the case where a model is restored from existing variables, i.e. in all other cases the variables are initialized randomly as usual. Then it initializes the one that are in the checkpoint with the values in the checkpoint and the ones that are missing in the checkpoint randomly. 

This is a kind of hacky solution for allowing additional variables that are not (re-)stored. In particular I need a variable to store the received rewards during training: https://github.com/ufal/neuralmonkey/pull/602/files#diff-58f4b514aef8636eabc281d307b2b61aR121
If you have a better idea how to solve this, I'd prefer to leave the learning_utils.py unchanged.

Reply:
We are now working on an experiment in which we need almost exactly the same functionality, and I think we could use this approach. Eventually though, it's likely that we'll need to come up with a different solution. (Namely, the Saver object can be initialized with a list of variables, so we could have multiple of these object for different parts of the graph.)

Reply:
Yes, that would be very practical!